### ISSUE TITLE
int + int * int wrongly interpreted as double

### ISSUE THREAD

**Issue by donatj** [scrutinizer-ci/scrutinizer#366]
https://scrutinizer-ci.com/g/capdig/myon/inspections/8818fbf4-e222-4d0e-81b0-ccf1a14f369c/issues/files/application/routes/www/exams/placement.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

![screen shot 2016-02-01 at 15 40 41](https://cloud.githubusercontent.com/assets/133747/12732273/6b0bbe48-c8fa-11e5-83c3-e402be4fa4d3.png)

I am being told that an [integer `time()`](http://us2.php.net/manual/en/function.time.php) + [an integer `\WM\ONE_MINUTE` ](https://github.com/henderjon/weights_and_measures/blob/master/src/weights_and_measures.php#L9) * 25 ***(integer)*** is a double.  

This is plainly wrong.

**Comment by FGM** [comment]
Not necessarily: if the value overflows PHP_INT_MAX (sometime in 2038 for 32-bit builds), it will be converted to a double, AIUI. Won't it ?

**Comment by donatj** [comment]
@FGM Any int could overflow though, should **every** int then be documented as `int|double`?

I set plenty of int documented params to ints elsewhere without this complaint, so I assume it's time() issue... So on a 64 bit system in the year 292,277,026,596 my int becomes a double and that's worth complaining about? I mean alright, 32 bit systems, which are all ready rare, we will need to worry about the 2038 bug, but I feel like 32 bit systems, let alone PHP will probably be out of style before then.
